streams/readable-byte-streams/general.any.js

```diff
@@ -2111,3 +2171,128 @@ test(() => {
   assert_throws_js(RangeError, () => new ReadableStream({ type: 'bytes' }, new HasSizeMethod()),
                    'constructor should throw when size on the prototype chain');
 }, 'ReadableStream constructor should not accept a strategy with a size defined if type is "bytes"');
+
+promise_test(async t => {
+  const stream = new ReadableStream({
+    pull: t.step_func(c => {
+      const view = new Uint8Array(c.byobRequest.view.buffer, 0, 1);
+      view[0] = 1;
+
+      c.byobRequest.respondWithNewView(view);
+    }),
+    type: 'bytes'
+  });
+  const reader = stream.getReader({ mode: 'byob' });
+
+  const result = await reader.read(new Uint8Array([4, 5, 6]));
+  assert_equals(result.done, false);
+
+  const view = result.value;
+  assert_equals(view.byteOffset, 0);
+  assert_equals(view.byteLength, 1);
+  assert_equals(view[0], 1);
+  assert_equals(view.buffer.byteLength, 3);
+  assert_array_equals([...new Uint8Array(view.buffer)], [1, 5, 6]);
+}, 'ReadableStream with byte source: respondWithNewView() with a smaller view');
+
+promise_test(async t => {
+  const stream = new ReadableStream({
+    pull: t.step_func(c => {
+      const view = new Uint8Array(c.byobRequest.view.buffer, 0, 0);
+
+      c.close();
+
+      c.byobRequest.respondWithNewView(view);
+    }),
+    type: 'bytes'
+  });
+  const reader = stream.getReader({ mode: 'byob' });
+
+  const result = await reader.read(new Uint8Array([4, 5, 6]));
+  assert_equals(result.done, true);
```

Comment:
assert_true strictly checks that the value is boolean, so you don't need to use assert_equals.